src/extension.ts

```diff
@@ -185,22 +140,13 @@ export function activate(context: vscode.ExtensionContext) {
     : vscode.CompletionItem
   {
     let item = new vscode.CompletionItem(args.entry.new_prefix);
-    item.insertText = "";
     item.sortText = new Array(args.index + 2).join('0');
-    item.range = new vscode.Range(args.position, args.position);
-    let arg: CommandArgs = {
-      old_prefix: '',
-      new_prefix: '',
-      old_suffix: args.entry.old_suffix,
-      new_suffix: args.entry.new_suffix,
-    };
-    item.command = {
-      arguments: [arg],
-      command: COMMAND_NAME,
-      title: "accept completion",
-    };
-    item.insertText = args.entry.new_prefix;
-    item.range = new vscode.Range(args.position.translate(0, -args.old_prefix.length), args.position);
+    
+    item.insertText = new vscode.SnippetString(args.entry.new_prefix)
+      .appendTabstop()
+      .appendText(args.entry.new_suffix);
+
+    item.range = new vscode.Range(args.position.translate(0, -args.old_prefix.length), args.position.translate(0, args.entry.old_suffix.length));
```

Comment:
the range was changed from the begging of the old prefix (`args.old_prefix.length`) to old suffix (`args.entry.old_suffix.length`)